lib/spack/llnl/util/multiproc.py

```diff
@@ -8,25 +8,48 @@
 than multiprocessing.Pool.apply() can.  For example, apply() will fail
 to pickle functions if they're passed indirectly as parameters.
 """
-from multiprocessing import Process, Pipe, Semaphore, Value
+import functools
+from multiprocessing import Semaphore, Value
 
-__all__ = ['spawn', 'parmap', 'Barrier']
+__all__ = ['Barrier']
 
 
-def spawn(f):
-    def fun(pipe, x):
-        pipe.send(f(x))
-        pipe.close()
-    return fun
+def deferred(func):
+    """Package a function call into something that can be invoked
+    at a later moment.
```

Comment:
> I'm not sure I like this as a decorator, because it's only obvious at function definition that something is deferred. NOT at the point of invocation.

My intent was to make it clear using function names. I used `detect_version_command` instead of `detect_version` to stress that the function call was returning a command to detect the version, not the version itself. I'm not opposed to this change, just saying I'll also remove the `_command` suffix as it doesn't make much sense if we don't use `deferred` as a decorator.

Reply:
Yeah, I saw that.  I think it's still clearer to write defer(f)(*args) -- the `_command` convention still wasn't clear to me and I had to go look up the function to make sure I knew what I was reading.

Reply:
Yeah, I saw that.  I think it's still clearer to write defer(f)(*args) -- the `_command` convention still wasn't clear to me and I had to go look up the function to make sure I knew what I was reading.

Reply:
Fixed in e72f8ebd11abcd9f0ba027c2709a10d949629411